src/cli/self_update.rs

```diff
@@ -100,21 +102,48 @@ pub(crate) const NEVER_SELF_UPDATE: bool = true;
 #[cfg(not(feature = "no-self-update"))]
 pub(crate) const NEVER_SELF_UPDATE: bool = false;
 
-#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[derive(Clone, Copy, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
 #[serde(rename_all = "kebab-case")]
 pub enum SelfUpdateMode {
+    #[default]
     Enable,
     Disable,
     CheckOnly,
 }
 
 impl SelfUpdateMode {
-    pub(crate) fn modes() -> &'static [&'static str] {
-        &["enable", "disable", "check-only"]
+    pub(crate) fn join(sep: &str) -> String {
+        let mut buf = String::new();
+        for mode in SelfUpdateMode::value_variants() {
+            if !buf.is_empty() {
+                buf.push_str(sep);
+            }
+            buf.push_str(mode.as_str());
+        }
+
+        buf
```

Comment:
You can use https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.join for this if you'd prefer, seems easier to parse to me :)

```suggestion
        Self::value_variants().iter().join(sep)
```

Reply:
Good call, changed it.